spec/components/discourse_hub_spec.rb

```diff
@@ -9,4 +9,57 @@
       expect(DiscourseHub.discourse_version_check).to eq(hub_response)
     end
   end
+
+  describe '#version_check_payload' do
+    it 'should include stats if Discourse Hub has not fetched stats since past 7 days' do
+      DiscourseHub.stats_fetched_at = 8.days.ago
+      json = JSON.parse(DiscourseHub.version_check_payload.to_json)
+
+      expect(json["topic_count"]).to be_present
+      expect(json["post_count"]).to be_present
+      expect(json["user_count"]).to be_present
+      expect(json["topics_7_days"]).to be_present
+      expect(json["topics_30_days"]).to be_present
+      expect(json["posts_7_days"]).to be_present
+      expect(json["posts_30_days"]).to be_present
+      expect(json["users_7_days"]).to be_present
+      expect(json["users_30_days"]).to be_present
+      expect(json["active_users_7_days"]).to be_present
+      expect(json["active_users_30_days"]).to be_present
+      expect(json["like_count"]).to be_present
+      expect(json["likes_7_days"]).to be_present
+      expect(json["likes_30_days"]).to be_present
+      expect(json["installed_version"]).to be_present
+      expect(json["branch"]).to be_present
+    end
+
+    it 'should not include stats if Discourse Hub has fetched stats in past 7 days' do
+      DiscourseHub.stats_fetched_at = 2.days.ago
+      json = JSON.parse(DiscourseHub.version_check_payload.to_json)
+
+      expect(json["topic_count"]).not_to be_present
+      expect(json["post_count"]).not_to be_present
+      expect(json["user_count"]).not_to be_present
+      expect(json["like_count"]).not_to be_present
+      expect(json["likes_7_days"]).not_to be_present
+      expect(json["likes_30_days"]).not_to be_present
+      expect(json["installed_version"]).to be_present
+      expect(json["branch"]).to be_present
+    end
+
+    it 'should not include stats if Discourse Hub has not fetched stats since past year and site setting send_anonymized_statistics_to_discourse is disabled' do
```

Comment:
This line is getting a little too long :P Generally you want to keep the lines to 80 chars so that we don't have to scroll while reading through code. Perhaps we can break this line down to:

```
describe 'when send_anonymize_stats is disabled' do
   describe 'when Discourse Hub has not fetched stats for the past year' do
     it 'should not..'
   end
end
```